spec/govuk_error/configuration_spec.rb

```diff
@@ -56,6 +56,32 @@
       end
     end
 
+    it "strips sensitive data from an error message" do
+      def error_message_with_rabbit_url(username, password)
+        "some info amqps://#{username}:#{password}@some.domain:1234 about this error"
+      end
```

Comment:
It's unusual to define a function inside a block like this and thus not the most idiomatic Ruby, a lambda would be more idiomatic but given the simplicity I think you can avoid a function.

```
base = "some info amqps://%<username>s:%<password>s@publishingmq.integration.govuk-internal.digital:5671 about this error"
not_scrubbed = sprintf(base, username: "username", password: "password")
scrubbed = sprintf(base, username: "xxx", password: "xxx")
```